### ISSUE TITLE
RCS1163 should not report event handler methods

### ISSUE THREAD

**Issue by JosefPihrt** [JosefPihrt/Roslynator#49]
RCS1163 should not report methods with signature **void MethodName(object, EventArgs)**.

**Comment by John-Welch** [comment]
An impressively quick fix! I see from the `SymbolUtility.IsEventHandlerMethod` method you've added that it's checking for methods with an `(object, EventArgs)` signature. I agree this will catch all conventional event-handling methods.

Another similar case in my code can be simplified to:
```C#
        private void AMethod()
        {
            Action<string> doSomething = IgnoreStringParameter;
            AnotherMethod(IgnoreStringParameter);
        }

        private void AnotherMethod(Action<string> notifier)
        {
            notifier("Hello");
        }

        private void IgnoreStringParameter(string param)
        {
        }
 ```
Here, removing the `param` parameter from the `IgnoreStringParameter` method would prevent the Method Group being assigned to the `Action<string>` delegate or passed as an argument to `AnotherMethod` and would result in a compile error.

I'm not familiar with the Roslyn API - is it reasonably easy to find the places that a method is used such as this and exclude such methods as well?

**Comment by JosefPihrt** [comment]
So you would like to exclude parameter(s) of methods that are not invoked and used only as a method group.

**Comment by John-Welch** [comment]
I think it might be any method that is ever used as a Method Group (even if it is invoked elsewhere) - anywhere that removing the arguments from the usage site is not possible so that RCS1163 cannot be avoided without suppressing it.

**Comment by JosefPihrt** [comment]
Ok, that makes sense. I'm moving it to #51.

**Comment by nvmkpk** [comment]
This is either not completely fixed or it was reintroduced. I am seeing this happen in my project.

**Comment by JosefPihrt** [comment]
@nvmkpk could you provide a code sample?

**Comment by nvmkpk** [comment]
I can provide a screenshot. I am working on a UWP application in Visual Studio 2017.

![image](https://user-images.githubusercontent.com/18663052/27345592-4b07f28c-55b8-11e7-8d8e-a9962a897747.png)

**Comment by nvmkpk** [comment]
I was able to reproduce using a new UWP application using 'Blank App (Universal Windows)' template for C#.
[App1.zip](https://github.com/JosefPihrt/Roslynator/files/1089038/App1.zip)

**Comment by JosefPihrt** [comment]
I was not able to open the solution because I do not have UWP component installed on my Visual Studio and the installation of that component failed. Nevertheless I discovered a bug in the analyzer RCS1163. It occurs when method declaration is declared within partial class and the reference (method group) is in other file than declaration. I've fix this bug. Could you install latest version (1.4.13) and let me know if the bug still exists?

**Comment by nvmkpk** [comment]
Yes, it is now fixed. Thank you.
